### ISSUE TITLE
How should we write content, together?

### ISSUE THREAD

**Issue by vandadnp** [crowdsourcebooks/theswiftbook#11]
In this issue we are going to talk about how we can all write content together.

I have placed headers in each .MD file. Each contributor can go into an .MD file and then start writing content that is missing for each one of the headers.

Could we put our heads together to come up with a way to make it easier to get contributions?

**Comment by vandadnp** [comment]
> If some has a lot of follower in social media, and introduce our project, it will be very helpful to engage in contribution.

I've currently contacted Natasha The Robot to talk about The Swift Book to her 16k followers on Twitter. She might be able to help with this if she sees fit.

> So, getting in touch with authors and asking to contribute this project would be good choice to make more active project.

I can have a chat with a few of the authors that I know of, and ask them to contribute either directly or indirectly, by getting others involved.

> To appreciate contributions, I think that it is a nice idea to include their name in documents.
That's a fantastic idea. I think we need to be able to define who a contributor is. I was thinking that we can mention the contributors' names and a quick sentence about them to introduce them to the readers.

However, this list can quite possibly become very long, if we think about contributors that will fix a typo and never contribute more. For this, I think we need to define what criteria a contributor has to be able to meet to be mentioned in the actual book as a contributor.

Off the top of my head, I can think of this rule:

**A contributor is someone who has  contributed 3 sizable pull requests in the past 3 months**

What are your thoughts?

**Comment by dakeshi** [comment]
Welcome to authors. Thank for your passion @vandadnp 👍 

> However, this list can quite possibly become very long, if we think about contributors that will fix a typo and never contribute more.

You're right. I didn't think of that. 

> A contributor is someone who has contributed 3 sizable pull requests in the past 3 months

 3 months? It would be good in the initial stage of the project. 

> Also, we need to think about adding regular contributions to the Crowd Source Books organization, so that they can perhaps directly work with the repository and also be able to manage issues and assign people, etc.

I agree of that. We need more passionate maintainers. 👪  
And I think that adopting [the code of conduct](http://contributor-covenant.org/) would be very helpful to make better community.

**Comment by vandadnp** [comment]
> 3 months? It would be good in the initial stage of the project.

That is the question. How long is enough? Or what should the rules be at all?

**Comment by vandadnp** [comment]
@dakeshi thanks for the link to the code of conduct. I have now added that to the project at https://github.com/crowdsourcebooks/theswiftbook/blob/master/README.md

**Comment by dakeshi** [comment]
@vandadnp thanks for adding the code of conduct. 

> I think we need to define what criteria a contributor has to be able to meet to be mentioned in the actual book as a contributor. 

I have thought about this issue for some days. And I think that it is not a good idea to distinguish contributors with some rules. First of all, it is very hard to make rules for `contributions`. Quality or quantity? Who is responsible to judge? Even though we know what is the good writing and code, it can make some troubles because we can not have same opinions all the time.

This is the community driven project. I think that it is very valuable, respectful and helpful activities to make active project ecosystem even though it is just improving one line of code or fixing one character. They are all contributors.

Without rules, I think that we could keep some main contributors in a page, such as `Special thanks page`. Because we know who is the most active writers and contributors.

**Comment by vandadnp** [comment]
I agree with you. This is very important. Everybody who contributres, in any form, should be mentioned as a contributor. When I mentioned that we need some rules, I was thinking about the main contributors who send sizable contributions, such as yourself for instance.

So should we do it like this?

* We will have a page where we give thanks to the main contributors of the project, perhaps with their photos if they need and their contact details and a sentence or two about them.

* We will have another page where we list all contributors, without regard to what their contributions have been?
